### PR TITLE
byron-proxy: DB can trace writes

### PR DESCRIPTION
**Pull request by avieth**
Requested so that the chain length can go into EKG

### CODE REVIEW

File: byron-proxy/src/exec/Logging.hs [hunk 1]
```diff
@@ -52,10 +53,18 @@ convertTrace trace = case trace of
                        , Monitoring.severity = sev
                        , Monitoring.privacy = Monitoring.Public
                        }
-        logContent = Monitoring.LogMessage text
+        logContent = content -- Monitoring.LogMessage text
         logObject  = Monitoring.LogObject name logMeta logContent
     f logObject
```
Comment (avieth) [inline byron-proxy/src/exec/Logging.hs]: @CodiePP is there something in iohk-monitoring that does this for me: gathers the thread id and current time the puts it in the appropriate places? Or is the user of iohk-monitoring expected to do this, as I've done here?

Comment (CodiePP) [reply byron-proxy/src/exec/Logging.hs]: `mkLOMeta`

File: byron-proxy/src/exec/Main.hs [hunk 2]
```diff
@@ -475,8 +477,11 @@ main = do
             , indexFilePath = bpoIndexPath bpo
             , slotsPerEpoch = epochSlots
             }
-      withDB dbc $ \db -> do
-        let server = runServer (Logging.convertTrace trace) (bpoServerOptions bpo) epochSlots db
-            client = runClient (Logging.convertTrace trace) (bpoClientOptions bpo) genesisConfig epochSlots db
+          -- Trace DB writes in such a way that they appear in EKG.
+          dbTracer = flip contramap (Logging.convertTrace trace) $ \(DB.DBWrite (SlotNo count)) ->
+            ("db", Monitoring.Info, Monitoring.LogValue "block count" (Monitoring.PureI (fromIntegral count)))
```
Comment (avieth) [inline byron-proxy/src/exec/Main.hs]: @CodiePP Is this enough to get an EKG gauge?

**Comment by avieth** [comment]
@disassembler I switched on EKG in the logger config and then let it write some blocks, but the gauge does not appear. @CodiePP am I doing it wrong?

**Comment by 365andreas** [comment]
> @disassembler I switched on EKG in the logger config and then let it write some blocks, but the gauge does not appear. @CodiePP am I doing it wrong?

No that is correct. 
For some reason we are not able to see this gauge in EKG. Alex and I were about to investigate it.
*BUT* we are convince that the variable is inserted EKG store since we can see it via Prometheus (http://localhost:12799/metrics) in the first line.

File: byron-proxy/src/exec/Main.hs (hunk 2 above)
Comment (365andreas) [reply byron-proxy/src/exec/Main.hs]: Yes.

**Comment by 365andreas** [comment]
You can see slot numder under `int:byron-proxy.db.block count` in [EKG](http://localhost:12789/)

**Comment by 365andreas** [comment]
rebased on master

**Pull request merged**